### PR TITLE
[producer] Fixed NPE in UpdateBuilderImpl::validateUpdateRecordIsSerializable

### PR DESCRIPTION
**Pull request by FelixGV**
Also expanded unit test.

## How was this PR tested?
Unit test.

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: internal/venice-common/src/main/java/com/linkedin/venice/writer/update/UpdateBuilderImpl.java [hunk 1]
```diff
@@ -136,7 +136,7 @@ Exception validateUpdateRecordIsSerializable(GenericRecord updateRecord) {
       Object unresolvedDatum = serializationException.getUnresolvedDatum();
       String unresolvedDatumType = unresolvedDatum instanceof GenericContainer
           ? ((GenericContainer) unresolvedDatum).getSchema().toString()
-          : unresolvedDatum.getClass().getSimpleName();
+          : unresolvedDatum == null ? "null" : unresolvedDatum.getClass().getSimpleName();
```
Comment (majisourav99) [inline internal/venice-common/src/main/java/com/linkedin/venice/writer/update/UpdateBuilderImpl.java:139]: nit: can we assign `unresolvedDatum == null ? "null" : unresolvedDatum.getClass().getSimpleName();` to a string var ? having 2 ternary operator in one statement kinda reads bad.

Comment (FelixGV) [reply internal/venice-common/src/main/java/com/linkedin/venice/writer/update/UpdateBuilderImpl.java:139]: Sure, I can rework it.

Comment (FelixGV) [reply internal/venice-common/src/main/java/com/linkedin/venice/writer/update/UpdateBuilderImpl.java]: Done in the latest commit.

**Review by nisargthakkar — APPROVED** [review]

**Review by majisourav99 — APPROVED** [review]
Thanks Felix

**Pull request merged**